### PR TITLE
feat(link) adding basic support for links on any nodes and resolver functionality

### PR DESCRIPTION
**Pull request by alexnm**
Adding support for the link sub-structure for each element node, which is transformed at resolver time into a link/navlink element wrapping the corresponding node

### CODE REVIEW

File: packages/teleport-types/src/uidl.ts [hunk 1]
```diff
@@ -189,6 +192,23 @@ export type UIDLStyleDefinitions = Record<string, UIDLStyleValue>
...
 export type UIDLEventDefinitions = Record<string, UIDLEventHandlerStatement[]>
 
+export type UIDLLinkURLDefinition = {
+  type: 'url'
+  options: { url: UIDLAttributeValue; newTab: boolean }
+}
+export type UIDLLinkPageDefinition = { type: 'navlink'; options: { routeName: string } }
+export type UIDLLinkMailDefinition = {
+  type: 'mail'
+  options: { mail: string; subject?: string }
```
Comment (JayaKrishnaNamburu) [inline packages/teleport-types/src/uidl.ts]: Hey, can we consider adding support for **body** too on the code-generators side alone. Down the line, we can just add it when needed on the UI.

File: packages/teleport-uidl-validator/src/uidl-schemas/component.json [hunk 2]
```diff
@@ -299,6 +299,12 @@
         "key": {"type": "string"},
         "dependency": {"$ref": "#/definitions/componentDependency"},
         "style": {"$ref": "#/definitions/styleDefinitions"},
+        "abilities": {
+          "type": "object",
+          "properties": {
+            "link": {"$ref": "#/definitions/linkDefinition"}
+          }
+        },
```
Comment (JayaKrishnaNamburu) [inline packages/teleport-uidl-validator/src/uidl-schemas/component.json]: TODO: Update project-schema at `https://docs.teleporthq.io/uidl-schema/v1/project.json` to support the new custom-code and abilities. UIDL's downloaded from playground still point to it. Nice to have both in sync :)

File: examples/uidl-samples/project.json [hunk 3]
```diff
@@ -526,15 +547,22 @@
...
               "type": "element",
               "content": {
                 "elementType": "image",
-                "attrs": {
-                  "url": "$props.url"
+                "abilities": {
+                  "link": {
+                    "type": "url",
+                    "options": {
+                      "url": "$props.link",
```
Comment (JayaKrishnaNamburu) [inline examples/uidl-samples/project.json]: Update validator to check for the url and link in abilities if they are being used from props. Right now if we run it throws validation errors for `link` and `url` even if they are being used. This component should throw only `url` from `propDefinitions` in validations.

Comment (alexnm) [reply examples/uidl-samples/project.json]: I added link.options.url in the traverse nodes function, it should be passed now when the validator is looking at all usages of UIDLAttributes. can you check again?

**Pull request merged**